calc.rb

```diff
@@ -0,0 +1,76 @@
+puts "The Calculator"
+puts "I'll take two numbers and perform a mathematical operation on them. Let's try."
+
+# First number
+puts "\nWhat is the first number?"
+num1 = gets.chomp
+
+# Handling invalid user input for first number
+if num1 == nil || (num1.to_i.to_s != num1 && num1.to_f.to_s != num1)
+	puts "Please type a valid first number. For example, type 3 instead of three."
+	num1 = gets.chomp
+end
+
+# Second number
+puts "What is the second number?"
+num2 = gets.chomp
+
+# Handling invalid user input for second number
+if num2 == nil || (num2.to_i.to_s != num2 && num2.to_f.to_s != num2)
+	puts "Please type a valid second number. For example, type 5 instead of five."
+	num2 = gets.chomp
+end
```

Comment:
This comment applies to both the validation blocks on line 9-12 and 18-22. Giving users an opportunity to re-enter data is an important aspect of good user interface design. If you changed the `if` keyword to `until`, you'd have a loop that would prompt the user for new input _until_ they provided something that passed your validation routine.

Your approach is solid and well thought-out; taking it one more step with an `until` (or `while`) loop would make it a slam-dunk.